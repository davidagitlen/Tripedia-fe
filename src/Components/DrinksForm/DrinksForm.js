import React, { useState } from 'react';
import './DrinksForm.scss';

const DrinksForm = ({ collapseForm, openForm, defaultForm}) => {
  const [form, toggleClicked] = useState({
    Cocktails: false,
    Beer: false,
    Wine: false,
    NonAlcoholic: false,
    SodaFactory: false,
    TikiBar: false
  });

  let mockProps = [
    'Cocktails',
    'Beer',
    'Wine',
    'Non-Alcoholic',
    'Soda Factory',
    'Tiki Bar'
  ];

  const checkBoxes = mockProps.map(checkBox => {
    let name = checkBox.replace(/-|\s/gi, '');
    return (
      <div 
        key={name}
        className='individual-checkbox__container'
      >   
      <input 
        className='drink__checkbox'
        type='checkbox'
        value={name}
        checked={form[name]}
        onChange={e => 
          toggleClicked({
            ...form, [e.target.value]: !form[e.target.value]
          })
        }
      />
      <label>{checkBox}</label>
      </div>
    );
  });

  if (openForm.DrinkForm) {
    return (
      <div className='drink-form__container'>
        <h2>- Drink -</h2>
        <div className='checkbox__container'>
          {checkBoxes}
        </div>
      </div>
    );
  } else {
    return ( 
      <div 
        className='drink-form-closed__container'
        onClick={e => 
          collapseForm({
            ...defaultForm,
            DrinkForm: !openForm.DrinkForm
        })
      }
      >
        <p>DrinkForm</p>
      </div>
    );
  }
}

export default DrinksForm;