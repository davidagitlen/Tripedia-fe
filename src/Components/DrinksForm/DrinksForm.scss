.drink-form__container {
  width: 100%;
  height: 8vh;
  background: lightpink;
}