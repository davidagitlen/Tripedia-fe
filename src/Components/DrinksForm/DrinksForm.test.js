import React, { createContext } from 'react';
import { shallow, mount} from 'enzyme';
import DrinksForm from './DrinksForm';

describe('DrinksForm', () => {

  it('should mock formContext and match a snapshot', () => {
    const contextValues = {
      origin: "",
      destination: "",
      tripSubmitted: false,
      attractions: {},
      accommodations: {},
      food: {},
      drinks: {},
      services: {},
      miscellaneous: {},
      selectedCategories: []
    };
    const FormContext = createContext(contextValues);
    const mockOpenForm = jest.fn();
    const mockCollapseForm = jest.fn();
    const mockDefaultForm = {
      StartForm: false,
      AttractionsForm: false,
      AccommodationsForm: false,
      FoodForm: false,
      DrinksForm: false,
      ServicesForm: false,
      SubmitTripForm: false
    };
    const mockFormObject = {
      ArtGalleries: [],
      WalkingTours: [],
      DayCamps: []
    }
    const wrapper = mount(
      <FormContext.Provider value={contextValues}>
        <DrinksForm
          collapseForm={mockCollapseForm}
          openForm={mockOpenForm}
          defaultForm={mockDefaultForm}
          formObject={mockFormObject}
        />
      </FormContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});