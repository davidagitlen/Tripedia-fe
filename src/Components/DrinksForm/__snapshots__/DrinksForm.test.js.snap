// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrinksForm should mock formContext and match a snapshot 1`] = `
<DrinksForm
  collapseForm={[MockFunction]}
  defaultForm={
    Object {
      "AccommodationsForm": false,
      "AttractionsForm": false,
      "DrinksForm": false,
      "FoodForm": false,
      "ServicesForm": false,
      "StartForm": false,
      "SubmitTripForm": false,
    }
  }
  formObject={
    Object {
      "ArtGalleries": Array [],
      "DayCamps": Array [],
      "WalkingTours": Array [],
    }
  }
  openForm={[MockFunction]}
>
  <div
    className="form-closed__container"
    onClick={[Function]}
  >
    <img
      alt="drinks"
      src="drink-pin.svg"
    />
    <p>
      - Drinks -
    </p>
  </div>
</DrinksForm>
`;
